### PR TITLE
GH#27901 - Remove default custom certificate for Ingress Controller

### PR DESCRIPTION
**Pull request by jboxman**
- https://github.com/openshift/openshift-docs/issues/27901

Preview: https://deploy-preview-30633--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator#nw-ingress-custom-default-certificate-remove_configuring-ingress

### CODE REVIEW

**Comment by jboxman** [comment]
@pamoedom, is the procedure only the command you provided? Or is more detail required?

Thanks!

**Comment by pamoedom** [comment]
Hi @jboxman, yes, is as simple as running that command (or manually edit the object and remove the line) and of course, **waiting for the clusteroperators to finish progressing**, it could take various minutes to propagate the changes cluster-wide.

Best Regards.

**Comment by jboxman** [comment]
@pamoedom, is there any example output of what that looks like that I can include in the documentation? Thanks!

**Comment by pamoedom** [comment]
@jboxman, you mean an example of the object including the line to be removed or an example of how it would look after the patching?

**Comment by jboxman** [comment]
@pamoedom, I mean the output from this:

> waiting for the clusteroperators to finish progressing, it could take various minutes to propagate the changes cluster-wide.

Is there a way to monitor the progress with the `oc` command?

Thanks!

**Comment by pamoedom** [comment]
@jboxman, got it, sorry. That part can be easily monitored with `oc get clusteroperators`.

**Comment by jboxman** [comment]
@bmcelvee, do you know someone from QE that could peek at this? Thanks!

**Comment by pamoedom** [comment]
@jboxman, I'm now part of QE Installer team, do you want me to double check this?

**Comment by jboxman** [comment]
@pamoedom sure!

**Comment by pamoedom** [comment]
@jboxman, I can see you have removed the character `$` from the patch command, please note that it will fail on that manner, the character is needed, here you have the proof:

[Without]
~~~
$ oc patch -n openshift-ingress-operator ingresscontrollers/default --type json -p '- op: remove\n  path: /spec/defaultCertificate'
error: unable to parse "- op: remove\\n  path: /spec/defaultCertificate": yaml: mapping values are not allowed in this context
~~~

[With]
~~~
$ oc patch -n openshift-ingress-operator ingresscontrollers/default --type json -p $'- op: remove\n  path: /spec/defaultCertificate'
ingresscontroller.operator.openshift.io/default patched
~~~

Regards.

**Comment by pamoedom** [comment]
Sure @jboxman, no hurries, just wanted to warn you about the necessary `$` character in the patch command, thanks for following up on this.

**Comment by bmcelvee** [comment]
Sounds like this is pretty well covered. Thanks! @lihongan, just wanted to make sure you see this too.

**Comment by jboxman** [comment]
@pamoedom, I've updated the procedure. I also updated the procedure for replacing the default certificate to include the more useful test to see if the certificate itself is different. Can you confirm? Thanks!

**Comment by pamoedom** [comment]
LGTM, thks @jboxman

File: modules/nw-ingress-custom-default-certificate-remove.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+.Verification
+
+* To confirm that the original cluster certificate is restored, enter the following command:
++
+[source,terminal]
+----
+$ echo Q |\
+  openssl s_client -connect console-openshift-console.apps.<domain>:443 -showcerts 2>/dev/null |\
+  openssl x509 -noout -subject -issuer -enddate
```
Comment (bobfuru) [inline modules/nw-ingress-custom-default-certificate-remove.adoc:36]: I'm not aware of us having any conventions about whether or not to add a space before a backslash to break up a long line of code in a block. My instincts tell me to add a space after the pipes in this example and in the next file on l80 to help avoid copy-paste confusion. 🤷

File: modules/nw-ingress-custom-default-certificate-remove.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+----
++
+where:
++
+--
+`<domain>`:: Specifies the base domain name for your cluster.
+--
++
+.Example output
+[source,text]
```
Comment (bobfuru) [inline modules/nw-ingress-custom-default-certificate-remove.adoc:46]: Do we care whether we should use `[source,text]` or `[source,terminal]` for example outputs from a terminal command? Seems our docs use both and I'm wondering if it renders any noticeable difference in the output. 

Oh, and I notice you changed `s/terminal/text` in the next file.

**Review by bobfuru — COMMENTED** [review]
Couple of comments for my education but overall this LGTM!

**Comment by jboxman** [comment]
@lihongan, can you confirm if this updated procedure is okay? Thank you!

File: modules/nw-ingress-custom-default-certificate-remove.adoc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+----
+subject=CN = *.apps.example.com
```
Comment (lihongan) [inline modules/nw-ingress-custom-default-certificate-remove.adoc:48]: after removing the custom certificates, the subject should be restore to `*.apps.<domain>`, the domain is the base domain name for you cluster.

**Comment by lihongan** [comment]
just one comment, others look good. Thank you @jboxman

File: modules/nw-ingress-custom-default-certificate-remove.adoc (hunk 3 above)
Comment (jboxman) [reply modules/nw-ingress-custom-default-certificate-remove.adoc:48]: @lihongan, great, thanks!

File: modules/nw-ingress-custom-default-certificate-remove.adoc (hunk 2 above)
Comment (jboxman) [reply modules/nw-ingress-custom-default-certificate-remove.adoc:46]: @bobfuru, using `text` will never accidentally apply any highlighting if `#` or `$` are present. Seems a sensible precaution.

File: modules/nw-ingress-custom-default-certificate-remove.adoc (hunk 1 above)
Comment (jboxman) [reply modules/nw-ingress-custom-default-certificate-remove.adoc:36]: @bobfuru, we don't, but we could...

**Comment by bergerhoffer** [comment]
This PR never got CPd to 4.9, so it hasn't gotten carried forward to 4.10-4.12. (And caused a conflict with another writer modifying this same file in main.

Checked in with @jboxman, who said it was good to CP forward. Doing so now!

**Pull request merged**